SignalServiceKit/src/Messages/Interactions/TSOutgoingMessage.m

```diff
@@ -455,11 +458,52 @@ - (OWSSignalServiceProtosDataMessage *)buildDataMessage
     return [[self dataMessageBuilder] build];
 }
 
-- (NSData *)buildPlainTextData
+- (void)addLocalProfileKeyIfNecessary:(OWSSignalServiceProtosContentBuilder *)contentBuilder
+                            recipient:(SignalRecipient *)recipient
+{
+    OWSAssert(contentBuilder);
+    OWSAssert(recipient);
+
+    OWSAssert(OWSProfilesManager.sharedManager.localProfileKey.length > 0);
+    BOOL shouldIncludeProfileKey = NO;
+
+    if ([self isKindOfClass:[OWSOutgoingSyncMessage class]]) {
+        // Always sync the profile key to linked devices.
+        shouldIncludeProfileKey = YES;
+    } else {
+        OWSAssert(self.thread);
+
+        // For 1:1 threads, we want to include the profile key IFF the
+        // contact is in the whitelist.
+        //
+        // For Group threads, we want to include the profile key IFF the
+        // recipient OR the group is in the whitelist.
+        if ([OWSProfilesManager.sharedManager isUserInProfileWhitelist:recipient.recipientId]) {
+            shouldIncludeProfileKey = YES;
+        } else if (self.thread.isGroupThread) {
+            TSGroupThread *groupThread = (TSGroupThread *)self.thread;
+            NSData *groupId = groupThread.groupModel.groupId;
+
+            if ([OWSProfilesManager.sharedManager isGroupIdInProfileWhitelist:groupId]) {
+                [contentBuilder setProfileKey:OWSProfilesManager.sharedManager.localProfileKey];
```

Comment:
based on the earlier code, I assume this should be:
```
shouldIncludeProfileKey = YES;
```

And then we only actually set the profile key in one place at the end.